File: java/hamming/src/main/java/Hamming.java
public class Hamming {
    static int compute(String strandA, String strandB) {
        int hammingDistance = 0;
        if (strandA.length() != strandB.length()) {
            throw new IllegalArgumentException("Strand lengths must be equal to calculate Hamming distance");
        }
        for (int i = 0; i < strandA.length(); i++) {
            int characterDistance = Character.compare(strandA.charAt(i), strandB.charAt(i));
            hammingDistance += (characterDistance != 0) ? 1 : 0;
        }
        return hammingDistance;
    }
}

File: lua/hamming/hamming.lua
local hamming = {}

function hamming.compute(strandA, strandB)
  local distance = 0

  if #strandA ~= #strandB then distance = -1
  else
    for i = 1, #strandA do
      if strandA:sub(i,i) ~= strandB:sub(i,i) then
        distance = distance + 1
      end
    end
  end

  return distance
end

return hamming
